docs/developer-docs/smart-contracts/advanced-features/randomness.mdx

```diff
@@ -51,28 +51,41 @@ In addition to the `raw_rand` method, Motoko offers a [Random module](/docs/curr
 
 </TabItem>
 
-<TabItem value="rust" label="rust">
+<TabItem value="rust" label="Rust">
 
 ```rust
 let (randomBytes,): (Vec<u8>,) = ic_cdk::api::call(Principal.management_canister(), "raw_rand", ()).await?;
 ```
 </TabItem>
 
-<TabItem value="typescript" label="typescript">
+<AdornedTab value={"typescript"} label="TypeScript" endAdornment={<BetaChip />}>
 
 ```typescript
-import { blob, Canister, ic, update } from 'azle';
-import { managementCanister } from 'azle/canisters/management';
-
-export default Canister({
-    randomBytes: update([], blob, async () => {
-        return await ic.call(managementCanister.raw_rand);
-    })
-});
+import { IDL, postUpgrade, query, update } from 'azle';
+
+let redeployed = false;
+
+export default class {
+    @postUpgrade([])
```

Comment:
I'm not sure why this example has postUpgrade and getRedeployed in it...seems this section is about randomness?

Reply:
Also I think this example should probably make a cross-canister call to the management canister to call raw_rand like the other examples do. Though you can use Math.random() to get some form of randomness, the actual VRF functionality I believe must come from raw_rand.

Reply:
It would look something like this, this is just the call:

```
await call('aaaaa-aa', 'raw_rand', {
    returnIdlType: IDL.Vec(IDL.Nat8)
});
```

Reply:
revised example